spec/myst/string_spec.mt

```diff
@@ -145,3 +145,98 @@ describe("String#to_s") do
     assert("\tone\ttwo".to_s == "\tone\ttwo")
   end
 end
+
+describe("String#chars") do
+  it("Returns a List with chars in a string") do
+    assert("abc".chars == ["a", "b", "c"])
+    assert("yay".chars == ["y", "a", "y"])
+  end
+end
+
+describe("String#downcase") do
+  it("Returns a lowercased version of itself") do
+    assert("HELLO".downcase == "hello")
+  end
+end
+
+describe("String#upcase") do
+  it("Returns a uppercased version of itself") do
+    assert("hello".upcase == "HELLO")
+  end
+end
+
+describe("String#chomp") do
+  it("Returns self with a newline at the end removed when no argument is provided") do
+    assert("Bob likes to fish\n".chomp == "Bob likes to fish")
+  end
+
+  it("Returns self with specified string removed from the end") do
+    assert("Fish does not like to be fish".chomp(" fish") == "Fish does not like to be")
+  end
+end
+
+describe("String#strip") do
+  it("Returns a new string with all leading and trailing whitespace removed") do
+    assert("   Hello \t".strip == "Hello")
+    assert((("  \t   \n    " * 7) + "Find me if you can" + ("\n" * 28)).strip == "Find me if you can")
+  end
+end
+
+describe("String#lstrip") do
+  it("Returns a new string with all leading whitespace removed") do
+    assert("   Hello \t".lstrip == "Hello \t")
+    assert("\t\tBye tabs".lstrip == "Bye tabs")
+  end
+end
+
+describe("String#rstrip") do
+  it("Returns a new string with all trailing whitespace removed") do
+    assert("   Hello \t".rstrip == "   Hello")
+    assert(" Nah\n\n\n".rstrip == " Nah")
+  end
+end
+
+describe("String#includes") do
+  it("Returns true if the string contains given string") do
+    assert("Bob fish? yeah".includes?("fish?") == true)
+    assert("Chess is awesome".includes?("Poker") == false)
+  end
+end
+
+describe("String#[]") do
+  it("Returns the char at the given index. A negative index is counted from the end of the string. \
+    Returns nil if the index falls outside of the string") do
+
+    str = "Hello"
+
+    assert(str[0] + str[-1] == "Ho")
+    assert(str[10] == nil)
+  end
```

Comment:
Splitting these into separate tests would be nice so that each test has a single assertion/responsibility. The `List#[]` spec would be a good example to base this on:

https://github.com/myst-lang/myst/blob/1fb8724b10a61570fabbd8d6c12eb858c2f0363a/spec/myst/list_spec.mt#L121-L152